pkg/odo/cli/url/describe.go

```diff
@@ -61,7 +67,42 @@ func (o *URLDescribeOptions) Validate() (err error) {
 
 // Run contains the logic for the odo url list command
 func (o *URLDescribeOptions) Run() (err error) {
-	if experimental.IsExperimentalModeEnabled() {
+	if pushtarget.IsPushTargetDocker() && experimental.IsExperimentalModeEnabled() {
+		client, err := lclient.New()
+		if err != nil {
+			return err
+		}
+		u, err := url.GetContainerURL(client, o.EnvSpecificInfo, o.url, o.EnvSpecificInfo.GetName())
+		if err != nil {
+			return err
+		}
+
+		if log.IsJSON() {
+			machineoutput.OutputSuccess(u)
+		} else {
+			tabWriterURL := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+			fmt.Fprintln(tabWriterURL, "NAME", "\t", "STATE", "\t", "URL", "\t", "PORT")
+			var urlString string
+			if u.Status.State == url.StateTypeNotPushed {
+				// to be consistent with URL for ingress and routes
+				// if not pushed, display URl as ://
+				urlString = "://"
+			} else {
+				urlString = u.Spec.Host + ":" + strconv.Itoa(u.Spec.ExternalPort)
+			}
+			// are there changes between local and cluster states?
+			outOfSync := false
+			fmt.Fprintln(tabWriterURL, u.Name, "\t", u.Status.State, "\t", urlString, "\t", u.Spec.Port)
+			if u.Status.State != url.StateTypePushed {
+				outOfSync = true
+			}
+			tabWriterURL.Flush()
+			if outOfSync {
+				fmt.Fprintf(os.Stdout, "\n")
+				fmt.Fprintf(os.Stdout, "There are local changes. Please run 'odo push'.\n")
+			}
+		}
+	} else if experimental.IsExperimentalModeEnabled() {
```

Comment:
Instead of adding an `else if` couldn't you do:

```go
if experimental.IsExperimentalModeEnabled() {
 if pushtarget.IsPushTargetDocker() {
   ...
     } else {
    ...
   }
}
```
?

Reply:
updated